File: RectangleArea/RectangleArea.cpp
#include<algorithm>
using namespace std;
class Solution {
public:
	int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
		int areA = (D - B) * (C - A);
		int areB = (H - F) * (G - E);
		int left = max(A, E);
		int right = min(C, G);
		int top = min(H, D);
		int bottom = max(B, F);
		int overlap = 0;
		if (left < right && top > bottom)
			overlap = (right - left) * (top - bottom);
		return areA + areB - overlap;
	}
};

File: RectangleArea/RectangleArea.py
class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """
        areA = (D-B) * (C-A)
        areB = (H-F) * (G-E)
        
        left = max(A, E)
        right = min(C, G)
        top = min(H, D)
        bottom = max(B, F)
        
        overlap = 0
        if left < right and top > bottom:
            overlap = (right-left) * (top-bottom)
        return areA+areB-overlap